Bootloader detection refactor

fauxpark opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

Figured out how to grab the vendor and product strings/IDs, and a more reliable method of obtaining the callout device for Caterina by looping through all the serial devices and matching its parent device's VID/PID with the one that was just plugged in. No more "No modem port found"!

![](https://media.discordapp.net/attachments/669700626858639360/790646488379555871/unknown.png)

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

nooges reviewed (changes requested):
Can you fix the code styling on the method declarations and calls, as per https://github.com/raywenderlich/objective-c-style-guide#methods?

Something like:
```objc
+ (void) eventMessageForDevice: (io_service_t) device withName: (NSString *) name connected: (BOOL) connected {
    [USB eventMessageForDevice: device withName: name withPort: nil connected: connected];
}
```

should look more like:
```objc
+ (void)eventMessageForDevice:(io_service_t)device withName:(NSString *)name connected:(BOOL)connected {
    [USB eventMessageForDevice:device withName:name withPort:nil connected:connected];
}
```

fauxpark commented:
I've always been kind of annoyed that there is no canonical style guide for Obj-C...

fauxpark commented:
- The preprocessor macros are gone from USB.m!
 - Implemented a slightly nicer method of obtaining the callout device on macOS for serial bootloaders
 - Sorted bootloader enums alphabetically, new entries should be inserted where necessary (nothing depends on the actual values)
 - Refactored the Windows side a little as well
